### ISSUE TITLE
Issue with the Dock icons in Ubuntu 22.04+

### ISSUE THREAD

**Issue by jhonny-oliveira** [deathkiller/jazz2-native#24]
Hi!

I'm packaging Jazz2 for Ubuntu 22.04+ and I'm planning to add it to the [XtraDeb](https://xtradeb.net) games portfolio. Maybe, even Debian, later on. Doing so, I noticed that the game icon doesn't show up properly in the Dock (defaulting to the Gnome generic one). 
You can find below the fix I applied. Perhaps, you could also apply it upstream.

```
Index: jazz2-1.5.0/Sources/project.desktop
===================================================================
--- jazz2-1.5.0.orig/Sources/project.desktop	2023-01-14 21:32:27.000000000 +0100
+++ jazz2-1.5.0/Sources/project.desktop	2023-01-17 00:35:04.971421789 +0100
@@ -5,6 +5,7 @@
 TryExec=@CPACK_EXECUTABLE_NAME@
 Exec=@CPACK_EXECUTABLE_NAME@
 Icon=@CPACK_EXECUTABLE_NAME@
+StartupWMClass=@NCINE_PREFERRED_BACKEND@-Application
 Terminal=false
 Type=Application
 Categories=Game;
```

Additionally, and because there is no section to place questions.

I'm building the game package with: 

```
		-DCMAKE_BUILD_TYPE=Release \
		-DNCINE_DOWNLOAD_DEPENDENCIES=OFF \
		-DNCINE_LINUX_PACKAGE=jazz2 \
		-DNCINE_WITH_GLEW=OFF \
		-DNCINE_VERSION_FROM_GIT=OFF \
		-DNCINE_LINKTIME_OPTIMIZATION=ON
```

This defaults to glfw. ARe these settings good for a release? Is it worth also building sdl or even qt5? Will the the end user loose anything if I don't?

I also noticed, that even if I pre-download the dependencies (Libs) they are not used. I assume the system dependencies are covering for this. Correct me if I'm wrong.

```
               cmake,
               libgl1-mesa-dev,
               libglfw3-dev,
               libsdl2-dev,
               libopenal-dev,
               libopenmpt-dev,
               zlib1g-dev
```

Last, this is the Debian package copyright I'm planning to attach. Is it OK with you or would you like to make some adjustments in the Files:* Copyright?

```
Format: http://www.debian.org/doc/packaging-manuals/copyright-format/1.0/
Upstream-Name: jazz2
Source: https://github.com/deathkiller/jazz2-native

Files: *
Copyright: Copyright (C) 2007 Free Software Foundation, Inc.
License: GPL-3.0+

Files: debian/*
Copyright: Copyright (C) 2023 Jhonny Oliveira <my email>
License: GPL-3.0+

```

Thank you!
Best regards,
Jhonny Oliveira

**Comment by deathkiller** [comment]
Hi! Thanks, I will change it, but I wonder if `StartupWMClass` can be just anything or it has to be `@NCINE_PREFERRED_BACKEND@-Application`.

I'm using [Discord](https://discord.gg/Y7SBvkD) for all discussions, but it's ok to discuss it here too.

I'm building it with following command on [GitHub Actions](https://github.com/deathkiller/jazz2-native/blob/master/.github/workflows/linux.yml#L67): `cmake -D CMAKE_BUILD_TYPE=Release -D NCINE_STRIP_BINARIES=ON -D NCINE_PREFERRED_BACKEND=$backend -D NCINE_ASSEMBLE_DEB=ON -D NCINE_ASSEMBLE_RPM=ON -D NCINE_WITH_GLEW=OFF`

`NCINE_ASSEMBLE_DEB` generates DEB package automatically. `NCINE_ASSEMBLE_RPM` generates RPM package which probably you don't need. `NCINE_PREFERRED_BACKEND` can be GLFW or SDL2 (QT5 is not supported). It seems SDL2 has better gamepad support. On the other hand, GLFW is more lightweight. But I haven't done any deep comparison, both works very similar for me. I turned `NCINE_LINKTIME_OPTIMIZATION` on by default recently, so you don't need to speficy it, but other options you mentioned are right. `NCINE_STRIP_BINARIES` can be used to strip debug information for smaller binaries.

And yes, system dependencies have higher priority than `Libs`, so you don't need to pre-download them separately. Also, I added Ogg Vorbis dependency recently, but I don't know name of the package.

I'm not sure about the copyright. As all code was written by me (except dependencies), it would be better this one:
```
Copyright: Copyright (C) 2016-2023 Dan R.
```

Thanks for including it in the repository and let me know if you have some other issues/questions.

**Comment by jhonny-oliveira** [comment]
The StartupWMClass needs to match the correct startup class, otherwise it doesn't work. I used "xprop WM_CLASS" to figure it out. I'm assuming the SDL startup class will match the same way. I did not test it.

I saw your pipelines. I inspired myself on them. :-)
NCINE_STRIP_BINARIES - The Debian packaging system will strip and build the dbgsym .deb automatically from it. So, this cannot be set if I want to make the debug symbols available to install.
NCINE_ASSEMBLE_DEB - I must use the Debian packaging method. Not needed.

Copyright updated.

Thank you!
